src/main/java/com/faforever/client/chat/TimePeriod.java

```diff
@@ -0,0 +1,16 @@
+package com.faforever.client.chat;
+
+public enum TimePeriod {
+  ALL_TIME,
+  LAST_YEAR,
+  LAST_MONTH;
+
+  public String getI18NKey() {
+    return switch (this) {
+      case ALL_TIME -> "userInfo.ratingHistory.allTime";
+      case LAST_YEAR -> "userInfo.ratingHistory.lastYear";
+      case LAST_MONTH -> "userInfo.ratingHistory.lastMonth";
+      default -> "";
+    };
+  }
```

Comment:
interesting. But did you know that enums like other classes in java can have instance variables and a constructor:
```suggestion
   ....
   LAST_MONTH("userInfo.ratingHistory.lastMonth")

  @Getter
  private final String i18nKey;
  
  TimePeriod(String i18nKey){
      this.i18nKey=i18nKey;
  }
```
Normally I would just annotate the class with @Value which makes variables final and private and also adds Constructor and Getter. But the @Value thing does not work on enums for whatever reasons.